Lib/asyncore.py

```diff
@@ -397,13 +394,14 @@ def close(self):
         self.del_channel()
         if self.socket is not None:
             try:
+                self._fileno = -1
```

Comment:
_fileno is initialized to None in ```__init__```, and set to None in ```del_channel```, so we should change it there.